### PR TITLE
Add .gitignore

### PR DESCRIPTION
**Pull request by lhpqaq**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,28 @@
+**/**.ninja
```
Comment (ram-mohan) [inline .gitignore:1]: Is this not sufficient?

```
# build output directories
build
build*

# downloaded dependencies
third_party/googletest
third_party/turbojpeg
third_party/benchmark
tests/data
```

**Review by ram-mohan — COMMENTED** [review]
These changes can be squashed with previous commit and can be force pushed.

File: .gitignore (hunk 1 above)
Comment (lhpqaq) [reply .gitignore]: Because it is uncertain whether the build directory is named 'build' 😂

**Review by ram-mohan — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (ram-mohan) [reply .gitignore]: I have suggested these changes basing on this, https://github.com/google/jpegli/blob/main/.gitignore and https://github.com/google/heir/blob/main/.gitignore, thank you making them

**Review by DichenZhang1 — APPROVED** [review]

**Pull request merged**